install/swiftly-install.sh

```diff
@@ -71,9 +115,57 @@ bold () {
     echo "$(tput bold)$1$(tput sgr0)"
 }
 
+# Fetch the list of required system dependencies from the apple/swift-docker
+# repository and attempt to install them using the system's package manager.
+#
+# $docker_platform_name, $docker_platform_version, and $package manager need
+# to be set before calling this function.
+install_system_deps () {
+    if [[ "$(id --user)" != "0" ]] && ! has_command sudo ; then
+        echo "Warning: sudo not installed and current user is not root, skipping system dependency installation."
+        return
+    elif ! has_command "$package_manager" ; then
+        echo "Warning: package manager \"$package_manager\" not found, skipping system dependency instllation."
+        return
+    fi
+
+    dockerfile_url="https://raw.githubusercontent.com/apple/swift-docker/main/nightly-main/$docker_platform_name/$docker_platform_version/Dockerfile"
+    dockerfile="$(curl --silent --retry 3 --location --fail $dockerfile_url)"
+    if [[ "$?" -ne 0 ]]; then
+        echo "Error enumerating system dependencies, skipping installation of system dependencies."
+    fi
+
+    # Find the line number of the RUN command associated with installing system dependencies.
+    beg_line_num=$(printf "$dockerfile" | grep -n --max-count=1 "$package_manager.*install" | cut -d ":" -f1)
+
+    # Starting from there, find the first line that doesn't have the same level of indentation.
+    relative_end_line_num=$(printf "$dockerfile" |
```

Comment:
I started using lowercase variable names for non-environment variables per https://stackoverflow.com/a/673940/16745421. In a later commit, I'll go back and update the existing variables to match this.